src/txmempool.cpp

```diff
@@ -305,15 +306,18 @@ void CTxMemPoolEntry::UpdateState(int64_t modifySize, CAmount modifyFee, int64_t
     }
 }
 
-CTxMemPool::CTxMemPool(const CFeeRate& _minRelayFee) :
+CTxMemPool::CTxMemPool(const CFeeRate& _minReasonableRelayFee) :
     nTransactionsUpdated(0)
 {
+    clear();
```

Comment:
This change produces a crash on osx.

```
jonasschnelli$ ./src/bitcoind --regtest
libc++abi.dylib: terminating with uncaught exception of type boost::exception_detail::clone_impl<boost::exception_detail::error_info_injector<boost::lock_error> >: boost: mutex lock failed in pthread_mutex_lock: Invalid argument
```

Stacktrace goes back to `cxx_global_var_initXX()`.
I think calling `LOCK()` from global var init (`CTxMemPool mempool(::minRelayTxFee);`) through this new `clear()` (which LOCKS mempool) is problematic.